### PR TITLE
Commit of first information theoretic measures

### PR DESCRIPTION
**Pull request by kzahedi**
- includes entropy measures
- includes mutual information measures
- include emperical estimations of joint distributions (1-3 variables)

Please take a look.

### CODE REVIEW

File: stat/it/probabilityestimators_test.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
+package it_test
+
+import (
+	"math"
+	"testing"
+
+	"github.com/kzahedi/gonum/stat/it"
```
Comment (kortschak) [inline stat/it/probabilityestimators_test.go]: Wrong package.

File: stat/it/entropy_test.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
+package it_test
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
```
Comment (kortschak) [inline stat/it/entropy_test.go]: This needs to be "golang.org/x/exp/rand" (this is the cause of the second failure).

File: stat/it/entropy_test.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	}
+}
+
+func TestEntropyChaoShen(t *testing.T) {
+	t.Log("Testing Chao-Shen Entropy")
+	r := 0.0
+	for i := 0; i < 100; i++ {
+		h := make([]int, 5000, 5000)
+		for j := 0; j < 5000; j++ {
+			h[j] = int(rand.Int63n(100))
```
Comment (kortschak) [inline stat/it/entropy_test.go]: rand.Intn(100), though we use per-function rand.Rand values to ensure test repeatability.

**Comment by btracey** [comment]
Can't we just make it `stat/entropy`? Then we have `entropy.ChaoShen`, `entropy.MLBC`, etc.

File: stat/it/entropy.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
+package it
```
[the remark this answers is not in the document]
Comment (kzahedi) [reply stat/it/entropy.go]: Done

File: stat/it/entropy.go [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+import (
+	"math"
+
+	"gonum.org/v1/gonum/mat"
+)
+
+// Entropy calculates the entropy of a probability distribution.
```
[the remark this answers is not in the document]
Comment (kzahedi) [reply stat/it/entropy.go]: I changed the comment and renamed the function to Shannon (see Brandon's comments below)

**Comment by kzahedi** [comment]
Thanks for the detailed comments. I will open a new pull request, in which I will address all the points.

**Comment by kortschak** [comment]
TBH, I'd prefer to keep working on this one. Migrating over to another PR means I need to re-review or review that in the context of this. It's much easier to just keep the information in one place.

**Pull request closed without merging**